repobee_sanitizer/_sanitize_repo.py

```diff
@@ -110,6 +111,21 @@ def sanitize_to_target_branch(
     Returns:
         List of errors if any errors are found, otherwise an empty list.
     """
+
+    if create_pr_branch:
+        repo = git.Repo(str(repo_path))
+
+        # If target branch is empty, create an empty commit or we cant create a new branch for PR
+        try:
+            repo.git.symbolic_ref("HEAD", f"refs/heads/{target_branch}")
+            repo.index.diff("HEAD")
+        except git.BadName as exc:
+            repo.git.commit("--allow-empty", "-m", "initial commit")
+
+        # Create and move to new branch before sanitizing
+        target_branch = "sanitizer-pull-request"
+        repo.create_head(target_branch)
```

Comment:
This is a bit bonkers :)

The gist of your problem is that you move HEAD to `refs/heads/{target_branch}`, and leave it there. So, you sanitize from the target branch ref, but from the state of the source branch.

The quick fix would be something like this:

```python
    if create_pr_branch:
        repo = git.Repo(str(repo_path))
        saved_ref_path = repo.head.ref.path
        
        [ do all the stuff ]

        repo.git.symbolic_ref("HEAD", saved_ref_path)
```

But you really don't need to fiddle with symbolic references here, all you need to do is to create swap out the target branch for an identical branch with a different name.

```python
    if create_pr_branch:
        repo = git.Repo(str(repo_path))
        new_target_branch = "sanitizer-pull-request"
        repo.git.branch(new_target_branch, target_branch)
        target_branch = new_target_branch
```

I'd argue that the target branch not existing really should be checked before-hand, otherwise the command has the side effect of just creating a branch with an empty commit. That seems pointless unless Sanitizer also pushes that empty branch to the platform and opens a PR from the PR branch. Not allowing a non-existing target branch seems reasonable.

I'd also argue that `sanitize_to_target_branch` is the wrong place to create the PR branch. Before this patch, `sanitize_to_target_branch` was a good abstraction, it did exactly what it's name implied. Now, however, it can either sanitize to the target branch, or to a PR branch based on the target branch. It would be better to create the PR branch before calling `sanitize_to_target_branch`, and simply pass the PR branch as the target branch.

Reply:
If the target branch is empty git will refuse to create a new branch from it.

So the alternatives i can find is:
* create empty commit
* sanitize to target_branch and create a pr branch from that after
    * I remember you saying we didnt want to do that during the meeting but i dont 100% remember why

Reply:
> If the target branch is empty git will refuse to create a new branch from it.

Which is totally reasonable. It doesn't make sense to create a PR into a branch that does not exist, and so it similarly does not make sense to run with `--create-pr-branch` if the target branch does not exist. Like I said, if you allow it, it make `--create-pr-branch` have the somewhat strange side effect of creating both the PR branch, and an _empty_ target branch. It's more likely that the user misspelled the target branch than that they actually wanted that to happen. Just don't allow that use case.

Pseudocode:

```python
if args.create_pr_branch and args.target_branch does not exist:
    raise plug.PlugError("that's dumb")
```

Reply:
Seems to be what you said with not going back to source branch. Thats on me, in my head i did that somewhere